hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/connector/WriteJdbcP.java

```diff
@@ -0,0 +1,172 @@
+import com.hazelcast.logging.ILogger;
+import com.hazelcast.util.concurrent.BackoffIdleStrategy;
+import com.hazelcast.util.concurrent.IdleStrategy;
+
+import javax.annotation.Nonnull;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.SQLNonTransientException;
+import java.util.ArrayList;
+import java.util.List;
+
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+import static java.util.concurrent.TimeUnit.SECONDS;
+
+/**
+ * Private API, use {@link SinkProcessors#writeJdbcP}.
+ */
+public final class WriteJdbcP<T> implements Processor {
+
+    private static final IdleStrategy IDLER =
+            new BackoffIdleStrategy(0, 0, MILLISECONDS.toNanos(1), SECONDS.toNanos(1));
+
+    private final DistributedSupplier<Connection> connectionSupplier;
+    private final DistributedBiConsumer<PreparedStatement, T> bindFn;
+    private final String updateQuery;
+
+    private ILogger logger;
+    private Connection connection;
+    private PreparedStatement statement;
+    private List<T> itemList = new ArrayList<>();
+    private int idleCount;
+
+    private WriteJdbcP(
+            @Nonnull String updateQuery,
+            @Nonnull DistributedSupplier<Connection> connectionSupplier,
+            @Nonnull DistributedBiConsumer<PreparedStatement, T> bindFn
+    ) {
+        this.updateQuery = updateQuery;
+        this.connectionSupplier = connectionSupplier;
+        this.bindFn = bindFn;
+    }
+
+    /**
+     * Private API, use {@link SinkProcessors#writeJdbcP}.
+     */
+    public static <T> ProcessorMetaSupplier metaSupplier(
+            @Nonnull String updateQuery,
+            @Nonnull DistributedSupplier<Connection> connectionSupplier,
+            @Nonnull DistributedBiConsumer<PreparedStatement, T> bindFn
+
+    ) {
+        return ProcessorMetaSupplier.preferLocalParallelismOne(() ->
+                new WriteJdbcP<>(updateQuery, connectionSupplier, bindFn));
+    }
+
+    @Override
+    public void init(@Nonnull Outbox outbox, @Nonnull Context context) {
+        logger = context.logger();
+        prepareStatement();
+    }
+
+    @Override
+    public void process(int ordinal, @Nonnull Inbox inbox) {
+        if (!reconnectIfNecessary()) {
+            return;
+        }
+        if (itemList.isEmpty()) {
+            inbox.drainTo(itemList);
+        }
+        try {
+            for (T item : itemList) {
+                bindFn.accept(statement, item);
+                statement.addBatch();
+            }
+            statement.executeBatch();
+            itemList.clear();
+        } catch (Exception e) {
+            if (e instanceof SQLNonTransientException ||
+                    e.getCause() instanceof SQLNonTransientException) {
+                throw ExceptionUtil.rethrow(e);
+            } else {
+                logger.warning("Exception during update", e.getCause());
+                idleCount++;
+            }
+        }
+        idleCount = 0;
```

Comment:
We miss test for reconnection, i think it doesn't work.
This line will always assign idleCount=0 in case of a recoverable exception